src/cairoUtilFuncGen/serialisation.ts

```diff
@@ -9,24 +10,44 @@ import {
   WarpLocation,
 } from '../utils/cairoTypeSystem';
 import { TranspileFailedError } from '../utils/errors';
-import { FELT252_INTO_BOOL } from '../utils/importPaths';
+import { CONTRACT_ADDRESS, FELT252_INTO_BOOL, OPTION_TRAIT } from '../utils/importPaths';
 
 export function serialiseReads(
   type: CairoType,
   readFelt: (offset: number) => string,
   readId: (offset: number) => string,
-): [reads: string[], pack: string, requiredImports: [string[], string][]] {
+): [
+  reads: string[],
+  pack: string,
+  requiredImports: { import: [string[], string]; isTrait?: boolean }[],
+] {
   const packExpression = producePackExpression(type);
   const reads: string[] = [];
-  const requiredImports: [string[], string][] = [];
+  const requiredImports: { import: [string[], string]; isTrait?: boolean }[] = [];
   const packString: string = packExpression
-    .map((elem: string | Read) => {
-      if (elem === Read.Felt) {
+    .map((elem: string | Read | (string | Read)[]) => {
+      if (elem instanceof Array) {
```

Comment:
What do you think if each of the elements of the returned packExpression is standarized in the way of `(elementType, dataRelated)` being `elementType` the enum value from `Read` and `dataRelated` an object were you store data needed to write correctly the reader.

For address could be something like `(Read.Address, {})` but for uN could be like `(Read.UN, {nBytes: 8})`.

That way we could do the conditional match of the elem in a more elegant way and avoid the assumption that if `elem` is an array, then it will always belong to an UN